src/J2N/Collections/Generic/EqualityComparer.cs

```diff
@@ -285,6 +289,10 @@ public override bool Equals(float x, float y)
             if (float.IsNaN(x) && float.IsNaN(y))
                 return true;
 
+            // If only one is NaN, they're not equal
```

Comment:
We are calling the `SingleToRawInt32Bits` method below to ensure these are not considered equal. It is also an edge case, so we don't need to optimize it.